File: Ejercicios-complejidad.py
v=[1,2,3,4]
# Ejercicio 1
def Intercambia (x,y):
    n = 0
    aux = x; n += 1
    x = y; n +=1
    y = aux; n += 1
    print ("La complejidad del ejercicio 1 es: ", n)
Intercambia (3,1)
# Ejericio 2
def maxi (x,y):
    n = 0
    if (x>=y):
        result = x
        n +=1
    else:
        result = y
        n +=1
    print ("La complejidad del ejercicio 2 es: ", n)
    return result
maxi (3,5)
# Ejercicio 3
def suma (v,tamaño):
    n = 0
    result = 0; n += 1
    for i in range (0,tamaño):
        result = result + v[i]
        n += 1
    print ("La complejidad del ejercicio 3 es: ", n)
    return result
suma (v,3)
# Ejercicio 4
def aSaber1 (v,tamaño):
    n = 0
    result = 0; n += 1
    i = 0; n += 1
    while (i < tamaño):
        result = result + v[i]; n += 1
        i += 1; n += 1
    print ("La complejidad del ejercicio 4 es: ", n)
    return result
aSaber1 (v,4)
# Ejercicio 5
def aSaber2 (v ,tamaño):
    n = 0
    result = 0; n += 1
    i = tamaño - 1; n += 1
    while (i >= 0):
        result = result + v[i]; n += 1
        i -= 1; n += 1
    print ("La complejidad del ejercicio 5 es: ", n)
    return result
aSaber2 (v,4)
# Ejercicio 6
def aSaber3 (v,tamaño):
    n = 0
    result = 0; n += 1
    i = tamaño - 1; n += 1
    while (i >= 0):
        result = result + v[i]
        i = i - 2; n += 1
    print ("La complejidad del ejercicio 6 es: ", n)
    return result
aSaber3 (v,4)

File: Ejercicios-complejidad.php
<?php
    // Ejercicio 1
    function Intercambia ($x,$y)
    {
        $n = 0;
        $aux = $x; $n++;
        $x = $y; $n++;
        $y = $aux; $n++;
        echo "La complejidad del ejercicio 1 es: ".$n."<hr>";
    }
    Intercambia (3,1);
    // Ejercicio 2
    function maxi ($x,$y)
    {
        $n = 0;
        if ($x >= $y)
        {
            $result = $x;
            $n++;
        }
        else
        {
            $result = $y;
            $n++;
        }
        echo "La complejidad del ejercicio 2 es: ".$n."<hr>";
        return $result;
    }
    maxi (3,5);
    // Ejercicio 3
    function suma ($v,$tamaño)
    {
        $n = 0;
        $result = 0; $n++;
        for($i = 0;$i <= $tamaño-1; $i++)
        {
            $result = $result + $v[$i];
            $n++;
        }
        echo "La complejidad del ejercicio 3 es: ".$n."<hr>";
        return $result;
    }
    suma ($v=[1,2,3],3);
    // Ejercicio 4
    function aSaber1 ($v,$tamaño)
    {
        $n = 0;
        $result = 0; $n++;
        $i = 0; $n++;
        while ($i < $tamaño)
        {
            $result = $result + $v[$i]; $n++;
            $i++; $n++;
        }
        echo "La complejidad del ejercicio 4 es: ".$n."<hr>";
        return $result;
    }
    aSaber1 ($v=[1,2,3],3);
    // Ejercicio 5
    function aSaber2 ($v,$tamaño)
    {
        $n = 0;
        $result = 0; $n++;
        $i = $tamaño - 1; $n++;
        while ($i >= 0)
        {
            $result = $result + $v[$i]; $n++;
            $i--; $n++;
        }
        echo "La complejidad del ejercicio 5 es: ".$n."<hr>";
        return $result;
    }
    aSaber2 ($v=[1,2,3],3);
    // Ejercicio 6
    function aSaber3 ($v,$tamaño)
    {
        $n = 0;
        $result = 0; $n++;
        $i = $tamaño - 1; $n++;
        while ($i >= 0)
        {
            $result = $result + $v[$i];
            $i = $i - 2;
            $n++;
        }
        echo "La complejidad del ejercicio 6 es: ".$n."<hr>";
        return $result;
    }
    aSaber3 ($v=[1,2,3],3);
?>